File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        val = (digits[-1]+1)%10
        carry = (digits[-1]+1)//10
        digits[-1]=val
        i=len(digits)-2
        while i>=0 and carry!=0:
            digits[i] += carry
            carry = digits[i]//10
            digits[i] %= 10
            i-=1
        if carry:
            digits.insert(0, carry)
        return digits

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        int n = digits.length;
        for(int i=n-1; i>=0; --i){
            digits[i] += carry;
            carry = digits[i]/10;
            digits[i] %= 10;
        }
        if(carry == 0) return digits;
        int[] ans = new int[n+1];
        ans[0] = carry;
        for(int i=1; i<n+1; ++i) ans[i] = digits[i-1];
        return ans;
    }
}